{{cookiecutter.project_name}}/notebooks/ModelValidation.py

```diff
@@ -0,0 +1,224 @@
+# TODO(required) : model_type
+model_type = get_model_type_from_recipe()
+
+# Custom metrics to be included. Set it to None if custom metrics are not needed.
+# Please refer to custom_metrics parameter in mlflow.evaluate documentation https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate
+# TODO(optional) : define custom metric function as necessary
+# TODO(optional) : custom_metrics
+custom_metrics = []
+
+# Define model validation rules. Set it to None if validation rules are not needed.
+# Please refer to custom_metrics parameter in mlflow.evaluate documentation https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate
+# TODO(optional) : validation_thresholds
+validation_thresholds = {}
+
+# A dictionary of additional configurations to supply to the evaluator.
+# Please refer to evaluator_config parameter in mlflow.evaluate documentation https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate
+# TODO(optional) : evaluator_config
+evaluator_config = {}
+
+# COMMAND ----------
+
+client.get_model_version(model_name, model_version).description
+
+# COMMAND ----------
+
+eval_result = None
+err = None
+
+def log_to_model_description(run, success):
+    run_info = run.info
+    run_link = "#mlflow/experiments/{0}/runs/{1}".format(run_info.experiment_id, run_info.run_id)
+    description = client.get_model_version(model_name, model_version).description
+    if success:
+        status = "succeeded"
+        emoji = ":white_check_mark:"
+    else:
+        status = "failed"
+        emoji = ":x:"
+    if description != "":
+        description += """
+            ---
+        """.replace(" ", "")
+    description += "### {0}Model validation {1}. Refer to [model validation run]({2}) for details\n".format(emoji, status, run_link)
+    client.update_model_version(
+        name=model_name,
+        version=model_version,
+        description=description
+    )
+
+# run evaluate
+with mlflow.start_run() as run, tempfile.TemporaryDirectory() as tmp_dir:
+
+    validation_thresholds_file = os.path.join(tmp_dir, "validation_thresholds.txt")
+    with open(validation_thresholds_file, "w") as f:
+        if validation_thresholds:
+            for metric_name in validation_thresholds:
+                f.write("{0:30}  {1}\n".format(metric_name, str(validation_thresholds[metric_name])))
+    mlflow.log_artifact(validation_thresholds_file)
+
+    try:
+        eval_result = mlflow.evaluate(
+            model=model_uri,
+            data=data,
+            targets=targets,
+            model_type=model_type,
+            evaluators=evaluators,
+            validation_thresholds=validation_thresholds,
+            custom_metrics=custom_metrics,
+            baseline_model=None if not enable_baseline_comparison else baseline_model_uri,
+            evaluator_config=evaluator_config,
+        )
+        metrics_file = os.path.join(tmp_dir, "metrics.txt")
```

Comment:
Can we log this as JSON or CSV instead of TXT? It's much easier to programmatically parse them, if a user wants to do some analysis.

Reply:
Sure. The metrics are also logged to the run. They can also get metrics instead of artifacts.

Reply:
Actually, the txt is to be more readable, it's something like the model evaluation output of mlflow recipe. If user want to get metric value they should get the metrics from the mlflow run.
What you think?

Reply:
Let's use txt and get some user feedback on this choice.